app/actions/CommonActions.scala

```diff
@@ -22,17 +27,43 @@ object CommonActions {
 
   object PreReleaseFeature extends ActionBuilder[Request] with LazyLogging {
     override def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]): Future[Result] = {
-      val cookie = request.cookies.get(qaCookie.name)
-
-      if (cookie.exists(_.value == qaCookie.value)) {
+      val qaCookie = request.cookies.get(passthroughCookie.name)
+      if (qaCookie.exists(_.value == passthroughCookie.value)) {
         block(request)
       } else {
-        cookie.foreach(_ => logger.warn("Invalid QA Cookie supplied"))
+        qaCookie.foreach(_ => logger.warn("Invalid QA Cookie supplied"))
         OAuthActions.AuthAction.authenticate(request, block)
       }
     }
   }
 
+  class RequestWithCookies[A](request: Request[A]) extends WrappedRequest[A](request) {
+    val testUserCookie = request.cookies.get(testUserCookieName)
```

Comment:
The 'test user cookie' is troubling to me. The problem is that there may be other parts of the system (for instance CAS) that aren't driven by the user's browser, and will _still_ need to know if a user is a test user or not, and they won't have access to the user's cookie. I guess we're driven to this because rendering the Checkout page requires us to know the list of Zuora services, which requires knowing which Touchpoint backend we're talking to. So some kind of session state (e.g. this cookie) is necessary. It gets troubling when the user-identity (ie first name/email) gets out of sync with the value of the testUserCookie.